pkg/interface/src/views/apps/links/components/lib/link-submit.js

```diff
@@ -1,153 +0,0 @@
-import React, { Component } from 'react';
-
-import { S3Upload } from '~/views/components/s3-upload';
-import { Spinner } from '~/views/components/Spinner';
-import { Icon } from "@tlon/indigo-react";
-
-export class LinkSubmit extends Component {
-  constructor() {
-    super();
-    this.state = {
-      linkValue: '',
-      linkTitle: '',
-      linkValid: false,
-      submitFocus: false,
-      disabled: false
-    };
-    this.setLinkValue = this.setLinkValue.bind(this);
-    this.setLinkTitle = this.setLinkTitle.bind(this);
-  }
-
-  onClickPost() {
-    const link = this.state.linkValue;
-    const title = this.state.linkTitle
-      ? this.state.linkTitle
-      : this.state.linkValue;
-      this.setState({ disabled: true });
-    this.props.api.links.postLink(this.props.resourcePath, link, title).then((r) => {
-      this.setState({
-        disabled: false,
-        linkValue: '',
-        linkTitle: '',
-        linkValid: false
-      });
-    });
-  }
-
-  setLinkValid(link) {
-    const URLparser = new RegExp(
-      /((?:([\w\d\.-]+)\:\/\/?){1}(?:(www)\.?){0,1}(((?:[\w\d-]+\.)*)([\w\d-]+\.[\w\d]+))){1}(?:\:(\d+)){0,1}((\/(?:(?:[^\/\s\?]+\/)*))(?:([^\?\/\s#]+?(?:.[^\?\s]+){0,1}){0,1}(?:\?([^\s#]+)){0,1})){0,1}(?:#([^#\s]+)){0,1}/
-    );
-
-    const validURL = URLparser.exec(link);
-
-    if (!validURL) {
-      const checkProtocol = URLparser.exec('http://' + link);
-      if (checkProtocol) {
-        this.setState({ linkValid: true });
-        this.setState({ linkValue: 'http://' + link });
```

Comment:
You dropped this logic, now if you type `4chan.org` it doesn't prepend `http://` to it. We want it agnostic to URI schema (gopher:// etc should work); but as it is in the PR right now 4chan has no URI schema at all and the link opens in the same tab as `http://localhost:9000/~link/4chan.org`.

Reply:
Hm good point. I don't know how it got dropped.

Reply:
Ok, I remember why I did this. I changed the regex because a URL like https://www.youtube.com/watch?v=ADiDaOVTixs&feature=youtu.be doesn't get registered and that's what I was testing. I just changed it back to what you had the regex as before and we can deal with the problem separately

Reply:
O_o weird